keps/sig-node/4786-direct-cgroup-stats-collection/README.md

```diff
@@ -0,0 +1,548 @@
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone /
+release*.
+
+-   [ ](R) Enhancement issue in release milestone, which links to KEP dir in
+    [kubernetes/enhancements](not the initial KEP PR)
+-   [ ](R) KEP approvers have approved the KEP status as `implementable`
+-   [ ](R) Design details are appropriately documented
+-   [ ](R) Test plan is in place, giving consideration to SIG Architecture and
+    SIG Testing input (including test refactors)
+    -   [ ] e2e Tests for all Beta API Operations (endpoints)
+    -   [ ](R) Ensure GA e2e tests meet requirements for
+        [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+    -   [ ](R) Minimum Two Week Window for GA e2e tests to prove flake free
+-   [ ](R) Graduation criteria is in place
+    -   [ ](R)
+        [all GA Endpoints](https://github.com/kubernetes/community/pull/1806)
+        must be hit by
+        [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+-   [ ](R) Production readiness review completed
+-   [ ](R) Production readiness review approved
+-   [ ] "Implementation History" section is up-to-date for milestone
+-   [ ] User-facing documentation has been created in [kubernetes/website], for
+    publication to [kubernetes.io]
+-   [ ] Supporting documentation—e.g., additional design documents, links to
+    mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The [cAdvisor] is still started and running behind the scene by the Kubelet
+after [KEP-2371] to gather system cgroup stats. Running cAdvisor brings in extra
+overhead and complexity. However, the stat collection of system cgroups is
+trivial to implement with a low overhead as there are only a handful of system
+cgroups.
+
+This KEP aims to eliminate the need to run cAdvisor with enablement of KEP-2371
+for better performance and simplicity.
+
+[cAdvisor]: https://github.com/google/cadvisor
+[KEP-2371]: https://github.com/kubernetes/enhancements/edit/master/keps/sig-node/2371-cri-pod-container-stats/README.md
+
+### Current Use of cAdvisor
+
+After enabling feature `PodAndContainerStatsFromCRI`, only
+[summary API][summary-api] invokes cAdvisor for stats of:
+
+*   Root filesystem
```

Comment:
Image filesystem also.

Reply:
Pls correct me if I'm wrong. I think imagefs is taken care of by CRI, no?

Reply:
It is not.

Reply:
did you refer to https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/stats/cri_stats_provider.go#L389 called by https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/server/stats/summary.go#L85?

Reply:
Looking at that code, we are using cadvisor for Availability and Capacity.

Reply:
Yes, but that [cadvisor](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/stats/cri_stats_provider.go#L456C28-L456C40) will be replaced to not use cadvisor